web-services/query/src/main/java/datawave/webservice/query/logic/composite/CompositeQueryLogic.java

```diff
@@ -109,14 +116,21 @@ public void run() {
                             results.add(last);
                         }
                     } catch (InterruptedException e) {
-                        log.warn("QueryLogic thread interrupted", e);
+                        // if this was on purpose, then just log and the loop will naturally exit
+                        if (interrupted) {
+                            log.warn("QueryLogic thread interrupted", e);
+                        }
+                        // otherwise something else unexpected happened
+                        throw new RuntimeException(e);
```

Comment:
Based on the comment, it seems that you want to put the `RuntimeException` in an `else` clause instead of always throwing a `RuntimeException` regardless of whether the `interrupted` flag is set.